src/PendingReason.php

```diff
@@ -295,4 +334,58 @@ public function cleanDBonPurge()
             ]
         );
     }
+
+    public static function getDefault()
+    {
+        $pending_reason = new PendingReason();
+        if (
+            $pending_reason->getFromDBByCrit([
+                'is_default' => 1,
+            ])
+        ) {
+            return $pending_reason;
+        }
+
+        return null;
+    }
+
+    public static function isDefaultPending()
+    {
+        $default_pending = self::getDefault();
+
+        return $default_pending && $default_pending->fields['is_pending_per_default'];
+    }
+
+    public function prepareInput(array $input)
+    {
+        if (isset($input['is_default']) && $input['is_default']) {
+            global $DB;
+
+            $DB->update(
+                $this->getTable(),
+                [
+                    'is_default' => 0,
+                ],
+                [
+                    'is_default' => 1,
+                ]
+            );
+        }
+
+        if (isset($input['is_pending_per_default']) && $input['is_pending_per_default']) {
+            $input['is_pending_per_default'] = $input['is_default'] ?? 0;
+        }
+
+        return $input;
+    }
+
+    public function add(array $input, $options = [], $history = true)
+    {
+        return parent::add($this->prepareInput($input), $options, $history);
+    }
+
+    public function update(array $input, $options = [], $history = true)
+    {
+        return parent::update($this->prepareInput($input), $options, $history);
+    }
```

Comment:
If you are redefining these methods to edit `$input`, you should redefine directly `prepareInputForAdd` and `prepareInputForUpdate`